src/main/java/pl/patrycja/ox/gameflow/TicTacToe.java

```diff
@@ -12,18 +15,85 @@
 
     /**
      * This method is called whenever we start a new game.
+     *
      * @param args is string's array which are put on the beginning of game
      */
     public static void main(String[] args) {
 
         boolean ifDemo = checkIfDemo(args);
-        UI ui = UIFactory.set(ifDemo);
 
-        Game game = new Game(ui);
+        Game game = setMode(args, ifDemo);
         game.play();
     }
 
+    private static Game setMode(String[] args, boolean ifDemo) {
+        GameSettings gameSettings;
+        if (ifDemo) {
+            Sequence sequence = new Sequence(args);
+            sequence.generateSequence();
+            gameSettings = getGameSettings(args);
+        } else {
+            gameSettings = GameSettings.builder()
+                    .ui(UIFactory.set(false))
+                    .build();
+        }
+        return new Game(gameSettings);
+    }
+
     private static boolean checkIfDemo(String[] args) {
-        return args.length > 0 && args[0].equals("demo");
+        return args.length == 3;
+    }
+
+    private static GameSettings getGameSettings(String[] args) {
+        GameSettings.GameSettingsBuilder gameSettings = GameSettings.builder();
```

Comment:
You do not need that, this is static so you have access wherever you want to use it.